parachains/runtimes/assets/statemint/tests/tests.rs

```diff
@@ -61,7 +69,166 @@ fn test_asset_xcm_trader_does_not_work_in_statemine() {
 
 			let asset: MultiAsset = (asset_multilocation, amount_needed).into();
 
-			// Buy weight should return an error, since asset trader not installed
+			// Make sure buy_weight does not return an error
+			assert_ok!(trader.buy_weight(bought, asset.into()));
+
+			// Drop trader
+			drop(trader);
+
+			// Make sure author(Alice) has received the amount
+			assert_eq!(
+				Assets::balance(1, AccountId::from(ALICE)),
+				ExistentialDeposit::get() + amount_needed
+			);
+
+			// We also need to ensure the total supply increased
+			assert_eq!(Assets::total_supply(1), ExistentialDeposit::get() + amount_needed);
```

Comment:
Why did this happen? I don't see a mint anywhere.

Reply:
few lines above there is a mint :),
```
			assert_ok!(Assets::mint(
				RuntimeHelper::<Runtime>::origin_of(AccountId::from(ALICE)),
				1,
				AccountId::from(ALICE).into(),
				ExistentialDeposit::get()
			));
```

https://github.com/paritytech/cumulus/pull/1910/files#diff-4bcab62dac08701a3e6b73eda0face53c0d8b3b921c802752fdb165ad6a994e8R38-R43

Reply:
doh! thanks